File: app/lib/models/permission.dart
class Permission {
  final int id;
  final String createdAt;
  final String updatedAt;
  final String name;

  const Permission({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.name,
  });

  factory Permission.fromJson(Map<String, dynamic> json) => Permission(
    id: json['id'] as int,
    createdAt: json['createdAt'] as String,
    updatedAt: json['updatedAt'] as String,
    name: json['name'] as String,
  );
}

File: backend/internal/model/permission.go
package model

import (
	"gorm.io/gorm"
	"time"
)

type Permission struct {
	ID        uint16 `gorm:"primaryKey"`
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt gorm.DeletedAt
	Name      string `gorm:"type:varchar(150);unique;check:length(trim(name)) >= 6"`
	// many-to-many (permission-to-role)
	Roles []Role `gorm:"many2many:role_permissions;foreignKey:ID;joinForeignKey:PermissionID;references:ID;joinReferences:RoleID"`
}
